make deps fails on centos6.5 on cmake install (no clang)

```
[+] downloading the cmake source
--2015-03-04 23:48:07--  https://osquery-packages.s3.amazonaws.com/deps/cmake-2.8.12.2.tar.gz
Resolving osquery-packages.s3.amazonaws.com... 54.231.18.41
Connecting to osquery-packages.s3.amazonaws.com|54.231.18.41|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 6068231 (5.8M) [application/x-gzip]
Saving to: “cmake-2.8.12.2.tar.gz”

100%[==================================================================================================================================================================================================================================================================================>] 6,068,231   3.05M/s   in 1.9s    

2015-03-04 23:48:10 (3.05 MB/s) - “cmake-2.8.12.2.tar.gz” saved [6068231/6068231]

[+] unpacking the cmake source
[+] building cmake
---------------------------------------------
CMake 2.8.12.2, Copyright 2000-2012 Kitware, Inc.
---------------------------------------------
Error when bootstrapping CMake:
Cannot find appropriate C compiler on this system.
Please specify one using environment variable CC.
See cmake_bootstrap.log for compilers attempted.

---------------------------------------------
Log of errors: /vagrant/.sources/cmake-2.8.12.2/Bootstrap.cmk/cmake_bootstrap.log
---------------------------------------------
make: *** [deps] Error 6
[vagrant@localhost vagrant]$ cat /vagrant/.sources/cmake-2.8.12.2/Bootstrap.cmk/cmake_bootstrap.log
Try: clang
Line: clang  cmake_bootstrap_12954_test.c -o cmake_bootstrap_12954_test
----------  file   -----------------------

#ifdef __cplusplus
# error "The CMAKE_C_COMPILER is set to a C++ compiler"
#endif

#include<stdio.h>

#if defined(__CLASSIC_C__)
int main(argc, argv)
  int argc;
  char* argv[];
#else
int main(int argc, char* argv[])
#endif
{
  printf("%d%c", (argv != 0), (char)0x0a);
  return argc-1;
}

------------------------------------------
/vagrant/.sources/cmake-2.8.12.2/bootstrap: line 512: clang: command not found
Test failed to compile
[vagrant@localhost vagrant]$ which clang
/usr/bin/which: no clang in (/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin:/home/vagrant/bin)
[vagrant@localhost vagrant]$ yum search clang
Loaded plugins: fastestmirror
Determining fastest mirrors
 * base: mirror.pac-12.org
 * epel: mirrors.syringanetworks.net
 * extras: mirrors.loosefoot.com
 * updates: mirror.team-cymru.org
testing-devtools-2-centos-6                                                                                                                                                                                                                                                                                         119/119
epel/pkgtags                                                                                                                                                                                                                                                                                         | 1.4 MB     00:00     
==================================================================================================================================================== N/S Matched: clang ====================================================================================================================================================
clang-devel.i686 : Header files for clang
clang-devel.x86_64 : Header files for clang
csmock-plugin-clang.noarch : csmock plug-in providing the support for Clang
clang.i686 : A C language family front-end for LLVM
clang.x86_64 : A C language family front-end for LLVM
clang-analyzer.noarch : A source code analysis framework
csclng.x86_64 : A compiler wrapper that runs Clang in background

  Name and summary matches only, use "search all" for everything.

[vagrant@localhost vagrant]$ rpm -qv clang
package clang is not installed
```

--- a/tools/provision.sh
+++ b/tools/provision.sh
@@ -468,12 +468,10 @@ function main() {
       fi
       popd
 
-      install_cmake
     elif [[ $DISTRO = "centos7" ]]; then
       package gcc
       package binutils
       package gcc-c++
-      install_cmake
     fi
 
     if [[ ! -f /usr/bin/cmake ]]; then
@@ -497,11 +495,12 @@ function main() {
     package rpm-devel
     package libblkid-devel
 
+    install_cmake
+    install_boost
+
     if [[ $DISTRO = "centos6" ]]; then
-      install_boost
       package libudev-devel
     elif [[ $DISTRO = "centos7" ]]; then
-      install_boost
       package systemd-devel
     fi
 

